### DISCUSSION TITLE
Can labelme annotate squares?

### DISCUSSION THREAD

**Discussion by zhang13359** [wkentaro/labelme#1517] [Ideas / Feature requests]
I didn't find this feature in labelme

**Comment by minewilliam** [comment]
No there isn't.

It's a bit of a hack, but you can use the circle tool instead. The shape has a center position and a point somewhere on the circumference. It's not hard to make a square from circle annotations; Take the distance between the two points and multiply it by 2 the to get the (interpreted) square's size.

<img width="561" height="468" alt="image" src="https://github.com/user-attachments/assets/c533dee1-0c9e-42c9-984f-0747053cc109" />

**Comment by wkentaro** [comment]
Maybe we could add a dialog or mode (e.g., shift+drag) to add square mode for the existing rectangle annotation.

**Reply by kancheng** [reply]
@wkentaro @zhang13359 @minewilliam 

I have completed this feature; please test it.

https://github.com/wkentaro/labelme/pull/1745

**Reply by kancheng** [reply]
@zhang13359 @wkentaro @minewilliam 

If needed, the feature can be tested in isolation by creating a clean conda environment and installing the branch directly via pip (e.g. from feature/draw-square).
This allows evaluation without affecting existing local setups.

```
conda create -n labelme-square python=3.10 -y
conda activate labelme-square
pip install git+https://github.com/kancheng/labelme.git@feature/draw-square
```

**Comment by kancheng** [comment]
@zhang13359 @wkentaro @minewilliam 

- https://github.com/wkentaro/labelme/pull/1748

Note: This implementation uses keyboard interaction to enable square-constrained rectangle drawing.

If needed, the feature can be tested in isolation by creating a clean conda environment and installing the branch directly via pip.  
This allows evaluation without affecting existing local setups.

![demoshift](https://github.com/user-attachments/assets/aae7f044-d6aa-4520-a8cb-94ece37af958)


```bash
conda create -n labelme-square python=3.10 -y
conda activate labelme-square
pip install git+https://github.com/kancheng/labelme.git@feature/square-rectangle
```
